src/Compilers/Core/Portable/Compilation/Expression.cs

```diff
@@ -46,7 +46,7 @@ public static ConstantValue SynthesizeNumeric(ITypeSymbol type, int value)
                 return SynthesizeNumeric(((INamedTypeSymbol)type).EnumUnderlyingType, value);
             }
 
-            return null;
+            return ConstantValue.Bad;
```

Comment:
What is the repro? Is "Bad" an appropriate value for the circumstance, or is there a case missing above that should create a valid value?

The change on this line is good, regardless.